### PR TITLE
Add TBB to functional tests

### LINKED ISSUE #4290: Port TBB functional testing support from `tbb-0.9.0` to `develop`
Once improvements that can be untangled and cleanly ported to existing functional tests have been merged into `develop` (see #4289, #4287 and #4286), let's port the remainder of the `tbb-0.9.0` branch as appropriate, and get tests passing.

### PR DESCRIPTION
**Pull request by rmol**
## Status

Ready for review

## Description of Changes

This is an update of the work done on the tbb-0.9.0 and tbb-0.12.0 branches to run our Selenium tests using the Tor Browser.

Fixes #4290.

## Testing

Run the test suite as usual. All tests should pass.

To confirm that Tor Browser is in use, connect to the Docker container's VNC port with `make -C securedrop docker-vnc` while the functional tests are running.

## Deployment

Most of the changes are restricted to test environments. 

### If you made changes to the server application code:

- [x] Linting (`make ci-lint`) and tests (`make -C securedrop test`) pass in the development container

### If you made non-trivial code changes:

- [x] I have written a test plan and validated it for this PR

### CODE REVIEW

**Comment by kushaldas** [comment]
First run sees two random failures, trying again.

```
platform linux2 -- Python 2.7.12, pytest-3.3.2, py-1.5.2, pluggy-0.6.0 -- /usr/bin/python
cachedir: tests/.cache
rootdir: /home/kdas/code/securedrop/securedrop/tests, inifile: pytest.ini
plugins: mock-1.7.1, cov-2.5.1
collected 21 items                                                                                                                                           

tests/functional/test_admin_interface.py::TestAdminInterface::test_admin_interface PASSED                                                              [  4%]
tests/functional/test_admin_interface.py::TestAdminInterface::test_admin_edits_hotp_secret FAILED                                                      [  9%]
tests/functional/test_admin_interface.py::TestAdminInterface::test_admin_deletes_user PASSED                                                           [ 14%]
tests/functional/test_admin_interface.py::TestAdminInterface::test_admin_updates_image PASSED                                                          [ 19%]
tests/functional/test_admin_interface.py::TestAdminInterface::test_ossec_alert_button PASSED                                                           [ 23%]
tests/functional/test_admin_interface.py::TestAdminInterface::test_admin_adds_admin_user PASSED                                                        [ 28%]
tests/functional/test_journalist.py::TestJournalist::test_journalist_verifies_deletion_of_one_submission_modal PASSED                                  [ 33%]
tests/functional/test_journalist.py::TestJournalist::test_journalist_uses_col_delete_collection_button_modal PASSED                                    [ 38%]
tests/functional/test_journalist.py::TestJournalist::test_journalist_uses_index_delete_collections_button_modal PASSED                                 [ 42%]
tests/functional/test_journalist.py::TestJournalist::test_journalist_interface_ui_with_modal FAILED                                                    [ 47%]
tests/functional/test_make_account_changes.py::TestMakeAccountChanges::test_admin_edit_account_html_template_rendering PASSED                          [ 52%]
tests/functional/test_source.py::TestSourceInterface::test_lookup_codename_hint PASSED                                                                 [ 57%]
tests/functional/test_source_notfound.py::TestSourceInterfaceNotFound::test_not_found PASSED                                                           [ 61%]
tests/functional/test_source_session_timeout.py::TestSourceSessions::test_source_session_timeout PASSED                                                [ 66%]
tests/functional/test_source_warnings.py::TestSourceInterfaceBannerWarnings::test_warning_appears_if_tor_browser_not_in_use PASSED                     [ 71%]
tests/functional/test_source_warnings.py::TestSourceInterfaceBannerWarnings::test_warning_appears_if_orbot_is_used PASSED                              [ 76%]
tests/functional/test_source_warnings.py::TestSourceInterfaceBannerWarnings::test_warning_high_security PASSED                                         [ 80%]
tests/functional/test_submit_and_retrieve_file.py::TestSubmitAndRetrieveFile::test_submit_and_retrieve_happy_path PASSED                               [ 85%]
tests/functional/test_submit_and_retrieve_file.py::TestSubmitAndRetrieveFile::test_source_cancels_at_login_page PASSED                                 [ 90%]
tests/functional/test_submit_and_retrieve_file.py::TestSubmitAndRetrieveFile::test_source_cancels_at_submit_page PASSED                                [ 95%]
tests/functional/test_submit_and_retrieve_message.py::TestSubmitAndRetrieveMessage::test_submit_and_retrieve_happy_path PASSED                         [100%]
```

**Comment by kushaldas** [comment]
```
platform linux2 -- Python 2.7.12, pytest-3.3.2, py-1.5.2, pluggy-0.6.0 -- /usr/bin/python
cachedir: tests/.cache
rootdir: /home/kdas/code/securedrop/securedrop/tests, inifile: pytest.ini
plugins: mock-1.7.1, cov-2.5.1
collected 21 items                                                                                                                                           

tests/functional/test_admin_interface.py::TestAdminInterface::test_admin_interface PASSED                                                              [  4%]
tests/functional/test_admin_interface.py::TestAdminInterface::test_admin_edits_hotp_secret PASSED                                                      [  9%]
tests/functional/test_admin_interface.py::TestAdminInterface::test_admin_deletes_user PASSED                                                           [ 14%]
tests/functional/test_admin_interface.py::TestAdminInterface::test_admin_updates_image PASSED                                                          [ 19%]
tests/functional/test_admin_interface.py::TestAdminInterface::test_ossec_alert_button PASSED                                                           [ 23%]
tests/functional/test_admin_interface.py::TestAdminInterface::test_admin_adds_admin_user PASSED                                                        [ 28%]
tests/functional/test_journalist.py::TestJournalist::test_journalist_verifies_deletion_of_one_submission_modal PASSED                                  [ 33%]
tests/functional/test_journalist.py::TestJournalist::test_journalist_uses_col_delete_collection_button_modal PASSED                                    [ 38%]
tests/functional/test_journalist.py::TestJournalist::test_journalist_uses_index_delete_collections_button_modal PASSED                                 [ 42%]
tests/functional/test_journalist.py::TestJournalist::test_journalist_interface_ui_with_modal PASSED                                                    [ 47%]
tests/functional/test_make_account_changes.py::TestMakeAccountChanges::test_admin_edit_account_html_template_rendering PASSED                          [ 52%]
tests/functional/test_source.py::TestSourceInterface::test_lookup_codename_hint PASSED                                                                 [ 57%]
tests/functional/test_source_notfound.py::TestSourceInterfaceNotFound::test_not_found PASSED                                                           [ 61%]
tests/functional/test_source_session_timeout.py::TestSourceSessions::test_source_session_timeout PASSED                                                [ 66%]
tests/functional/test_source_warnings.py::TestSourceInterfaceBannerWarnings::test_warning_appears_if_tor_browser_not_in_use PASSED                     [ 71%]
tests/functional/test_source_warnings.py::TestSourceInterfaceBannerWarnings::test_warning_appears_if_orbot_is_used PASSED                              [ 76%]
tests/functional/test_source_warnings.py::TestSourceInterfaceBannerWarnings::test_warning_high_security PASSED                                         [ 80%]
tests/functional/test_submit_and_retrieve_file.py::TestSubmitAndRetrieveFile::test_submit_and_retrieve_happy_path PASSED                               [ 85%]
tests/functional/test_submit_and_retrieve_file.py::TestSubmitAndRetrieveFile::test_source_cancels_at_login_page PASSED                                 [ 90%]
tests/functional/test_submit_and_retrieve_file.py::TestSubmitAndRetrieveFile::test_source_cancels_at_submit_page PASSED                                [ 95%]
tests/functional/test_submit_and_retrieve_message.py::TestSubmitAndRetrieveMessage::test_submit_and_retrieve_happy_path PASSED                         [100%]
```

Second run, all green locally.

**Comment by kushaldas** [comment]
I ran the Python3 based tests 5 times, twice `test_admin_edits_hotp_secret` failed. This one seems to be following the history of the TBB branch :( 

Otherwise things look great. I will go through the text diff now.

**Comment by zenmonkeykstop** [comment]
@kushaldas how would you feel about marking `test_admin_edits_hotp_secret` as `xfail`, and waiting until #3844  is in and the flaky plugin can be added before un `xfail`ing it?

**Comment by kushaldas** [comment]
>@kushaldas how would you feel about marking test_admin_edits_hotp_secret as xfail, and waiting until #3844 is in and the flaky plugin can be added before un xfailing it?

That can one way, right now @rmol is fixing a few other old artifacts.

**Comment by redshiftzero** [comment]
@kushaldas: before we have @rmol do another rebase which is tricky given the size of this PR, want to do a review of the diff to see if there are any major issues that should be cleaned up prior to merge into `develop`?

File: changelog.md [hunk 1]
```diff
@@ -34,7 +34,7 @@
 ### Operations
 
 * Ensured WiFi related packages are not installed on Xenial on upgrade (#4163)
-* Try harder to attach to a `tmux` session on upgrade (#4221) 
+* Try harder to attach to a `tmux` session on upgrade (#4221)
```
Comment (redshiftzero) [inline changelog.md]: accidentally committed?

File: .isort.cfg [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[settings]
+line_length = 100
+indent = 4
+force_single_line = true
+case_sensitive = true
+# force_sort_within_sections = true
+default_section = THIRDPARTY
+sections = FUTURE,STDLIB,THIRDPARTY,FIRSTPARTY,LOCALFOLDER
+known_first_party = db,journalist_app,source_app,models,sdconfig,tests
```
Comment (redshiftzero) [inline .isort.cfg]: accidentally committed?

File: install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/files/changelog-trusty [hunk 3]
```diff
@@ -12,13 +12,13 @@ securedrop-app-code (0.12.2+trusty) trusty; urgency=medium
 
 securedrop-app-code (0.12.1+trusty) trusty; urgency=medium
 
-  * See changelog.md 
+  * See changelog.md
```
Comment (redshiftzero) [inline install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/files/changelog-trusty]: these changelog files I think are accidentally included here

File: securedrop/tests/pageslayout/test_source.py [hunk 4]
```diff
@@ -31,11 +31,6 @@ def test_index(self):
         self._source_visits_source_homepage()
         self._screenshot('source-index.png')
 
-    def test_index_javascript(self):
```
Comment (redshiftzero) [inline securedrop/tests/pageslayout/test_source.py]: removed due to flakiness?

File: install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/files/changelog-trusty (hunk 3 above)
Comment (redshiftzero) [reply install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/files/changelog-trusty]: [I've removed these whitespace changes in the changelog and rebased and squashed my commit]

File: .isort.cfg (hunk 2 above)
Comment (redshiftzero) [reply .isort.cfg]: [I've removed this file and rebased and squashed my commit, if we want to use isort we should open an issue and do in a separate PR, else you should add this file to your global gitignore]

**Pull request merged**